prefix-hex/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! The `hex-prefix` crates offers encoding and decoding of hex strings with a `0x` prefix.
+//!
+//! Its API aims to mimic that of the [`hex`](https://docs.rs/hex/latest/hex/) crate, which we also use internally.
+//!
+//! This crate is compatible with the hex encoding rules of the [Ethereum RPC API](https://eth.wiki/json-rpc/API#hex-value-encoding).
+
+#![no_std]
+
+extern crate alloc;
+
+mod data;
+mod error;
+#[cfg(feature = "primitive-types1")]
+mod primitive_types;
+
+use alloc::string::String;
+
+pub use error::Error;
+
+/// Tries to decode an hexadecimal encoded string with a `0x` prefix.
+pub trait FromHexPrefix: Sized {
+    /// Tries to decode an hexadecimal encoded string with a `0x` prefix.
+    fn from_hex_prefix(hex: &str) -> Result<Self, Error>;
+}
+
+// TODO: Maybe introduce `handle_error` with `#[cold]` attribute.
+fn strip_prefix(hex: &str) -> Result<&str, Error> {
+    if let Some(hex) = hex.strip_prefix("0x") {
+        Ok(hex)
+    } else if hex.len() < 2 {
+        Err(Error::InvalidStringLength)
+    } else {
+        let mut chars = hex.chars();
+        // Safety the following two operations are safe because we checked for the `hex.len()` in a previous branch.
+        let c0 = chars.next().unwrap();
+        let c1 = chars.next().unwrap();
```

Comment:
I know it's nit-picky (but we weren't the Bee team if there wouldn't be some bike-shedding in the review process), but the below calls aren't unsafe methods, but just unwraps, that can panic. Therefore my current preference is to document this with:
```rust
// Panic: the following two operations cannot panic because we checked ....
```